### PR TITLE
Feat/sso user info

### PR DESCRIPTION
**Pull request by patricksimonian**
1. check if user is Authenticated from SSO login
2. check if user is Authorized from Reggie API
3. redirect based on user info

### CODE REVIEW

**Comment by ShellyXueHan** [comment]
@patricksimonian got the async action working, also fixed the conflicts!

File: public/manifest.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "short_name": "React App",
-  "name": "Create React App Sample",
+  "short_name": "Reggie Web",
+  "name": "BC Gov SSO Registratin Website",
```
Comment (patricksimonian) [inline public/manifest.json]: Registration

File: src/components/UI/Header.js [hunk 2]
```diff
@@ -40,4 +41,17 @@ Header.propTypes = {
...
+  };
+};
+
+const mapDispatchToProps = dispatch => {
+  return {};
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
```
Comment (patricksimonian) [inline src/components/UI/Header.js]: if you don't have any actioins to dispatch you can leave this argument as `null`

File: src/reducers/index.js [hunk 3]
```diff
@@ -20,20 +20,82 @@
...
+import { AUTHENTICATION, AUTHORIZATION } from '../actions/actionTypes';
 
-const authentication = (state = { isAuthenticated: false }, action) => {
+const authentication = (state = { isAuthenticated: false, email: null }, action) => {
   switch (action.type) {
     case AUTHENTICATION.SUCCESS:
-      return { isAuthenticated: true };
+      return {
+        ...state,
+        ...{
```
Comment (patricksimonian) [inline src/reducers/index.js:30]: don't need to spread, 
```js
{
   ...state,
   isAuthenticated: true,
   email: implicitAuthManager.idToken.data.email,
}
```

File: src/reducers/index.js [hunk 4]
```diff
@@ -20,20 +20,82 @@
...
+          isAuthorized: false,
+          authorizationStarted: true,
+          userInfo: action.payload.userInfo,
+          ssoGroup: action.payload.ssoGroup,
+        },
+      };
+    case AUTHORIZATION.ERROR:
+      return {
+        ...state,
```
Comment (patricksimonian) [inline src/reducers/index.js:84]: perhaps we can look at integrating some 3rd party helpers for these immutable state changes

**Review by patricksimonian — COMMENTED** [review]
maybe we should make a few changes here

**Pull request merged**